test/concerns/scopes_test.rb

```diff
@@ -6,6 +6,9 @@ def test_scopes
       # Roots assertion
       assert_equal roots.map(&:first), model.roots.to_a
 
+      # Leaves assertion
+      assert_equal model.all.select(&:is_childless?), model.leaves.order(:id).to_a
```

Comment:
well, this does look like it is bringing back nodes from 2 levels deep.

ASIDE (nothing to do with you or this PR):
The way the tests are organized, it sure makes it hard to know where different elements are tested.